sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -145,13 +145,14 @@ public MetricData collectAndReset(
     try {
       try {
         metricUpdater.run();
-      } catch (RuntimeException e) {
+      } catch (Throwable e) {
         logger.log(
             Level.WARNING,
             "An exception occurred invoking callback for instrument "
                 + getMetricDescriptor().getName()
                 + ".",
             e);
+        propagateIfFatal(e);
```

Comment:
Let's put this above the log statement. We're already hosed so the log statement may just throw again in a weird way.

Reply:
+1, need to avoid allocations in the event you catch OOM.